myia/ir/manager.py

```diff
@@ -730,9 +731,12 @@ def limit(x):
 
         acq = OrderedSet()
         for node in nodes:
-            new_nodes = OrderedSet(dfs(node, succ_deeper, limit))
-            self.all_nodes |= new_nodes
-            acq |= new_nodes
+            acq |= OrderedSet(dfs(node, succ_deeper, limit))
+
+        if self.check_opaque:
+            acq = [n for n in acq if not self.check_opaque(n)]
```

Comment:
I'm worried that this can balloon for large graph clusters.

It might be better to add a limit on the dfs that respects check_opaque().

Reply:
You're right, I should've put this in the `limit` function that's defined right above this. I made a commit to fix this, we'll see if everything passes.